lib/logstash/inputs/google_pubsub.rb

```diff
@@ -254,6 +257,7 @@ def run(queue)
           if msg.key?("message") and msg["message"].key?("data")
             decoded_msg = Base64.decode64(msg["message"]["data"])
             @codec.decode(decoded_msg) do |event|
+              event.set("messageId", msg["message"]["messageId"]) if @include_message_id
```

Comment:
@andyshinn what do you think about pulling out the whole `msg["message"]` here and sticking it under something like `@message` so it would close #7?

Reply:
I think I tried that and ran into decoder errors... But let's leave this open for another week and I'll give that a try.

One reason I can think to have these both separate options is the data size increase. In our case, we are potentially going to be working with hundreds of thousands of messages and the difference in size at that scale could be 2-3x versus just the `messageId`.

Reply:
Actually, now that I think about it, the amount of data doesn't really matter much for the output side (where we care most) since we could just do some filtering at Logstash to grab what we want.

Reply:
I think there's also a [`@metadata` field](https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#metadata) that could be used. From my understanding, Logstash removes it before sending it to an output plugin, but you can use it in filters. That might let you keep the whole message, pull out the pieces you need in the pipeline and not even need a flag to enable it in the plugin. 

That's assuming I'm understanding it right... :-)

Reply:
Yep! I reworked this a bit to go that route and I actually like it. Seems a bit more flexible overall. I put two examples in the PR as well for how I tested it.